File: 1911-maximum-alternating-subsequence-sum/1911-maximum-alternating-subsequence-sum.java
class Solution {
    public long maxAlternatingSum(int[] nums) {
        int n = nums.length;
        long evenMax = 0, oddMax = 0;
        for(int i = 0; i < n; i++){
            long odd = oddMax;
            oddMax = Math.max(oddMax,evenMax - nums[i]); 
            evenMax = Math.max(Math.max(nums[i],evenMax),odd + nums[i]);
        }
        
        return evenMax;
    }
}

File: 1911-maximum-alternating-subsequence-sum/1911-maximum-alternating-subsequence-sum.py
class Solution:
    def maxAlternatingSum(self, nums: List[int]) -> int:
        
        n = len(nums)
        even  = 0
        odd = 0
        for i in range(1,n+1):
            temp = even
            even = max(even,odd - nums[i-1])
            odd = max(odd,even + nums[i-1])
            
        return max(even,odd)